standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1525,39 +1651,42 @@ public void markRefused(CompactionInfo info) throws MetaException {
     updateStatus(info);
   }
 
-
   @Override
   @RetrySemantics.CannotRetry
-  public void setCleanerRetryRetentionTimeOnError(CompactionInfo info) throws MetaException {
+  public void insertOrSetCleanerRetryRetentionTimeOnError(CompactionInfo info) throws MetaException {
```

Comment:
why do we need protective insert or set? don't we know if an entry is present or not?

Reply:
We know whether the entry is present or not beforehand. I just changed the name of this API to indicate that we either insert or update the compaction queue entry.

Reply:
closing this as dup, we already have a comment to simplify API